lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -206,6 +211,42 @@ protected function _getPortableTableColumnDefinition($tableColumn)
         return $column;
     }
 
+
+    /**
+     * Return Doctrine/Mysql-compatible column default values for MariaDB 10.2.7+ servers.
+     *
+     * - Since MariaDb 10.2.7 column defaults stored in information_schema are now quoted
+     *   to distinguish them from expressions (see MDEV-10134 for a what is an expression).
+     * - CURRENT_TIMESTAMP, CURRENT_TIME, CURRENT_DATE are stored in information_schema
+     *   as current_timestamp(), currdate(), currtime()
+     * - Literal escaping is normalized in information schema (store "''" instead of "\'")
+     *
+     * @link https://mariadb.com/kb/en/library/information-schema-columns-table/
+     * @link https://jira.mariadb.org/browse/MDEV-13132
+     *
+     * @param null|string $columnDefault default value as stored in information_schema for MariaDB >= 10.2.7
+     */
+    private function getMariaDb1027ColumnDefault(MariaDb102Platform $platform, ?string $columnDefault): ?string {
+        if ($columnDefault === 'NULL' || $columnDefault === null) {
```

Comment:
Yes depending if the column is nullable. But from what I've tested sometimes from the type as well. I guess there still a bug somewhere in maria. I'll check on next version to be sure, but testing both is safer to me.

Reply:
Yes depending if the column is nullable. But from what I've tested sometimes from the type as well. I guess there still a bug somewhere in maria. I'll check on next version to be sure, but testing both is safer to me.

Reply:
> I'll check on next version to be sure, but testing both is safer to me.

I'm fine with this approach as long as there's a _functional_ test which covers both cases. Also, it wouldn't hurt if the logic like this had a comment containing a reference to documentation explaining why exactly it's needed. Otherwise, it requires research to understand why the logic looks like this.

Reply:
@morozov, 100% agree. I created an issue for mariadb : https://jira.mariadb.org/browse/MDEV-14053 and added a comment in the latest commit (with the mariadb issue link too). 
This is an edge case and happen only after upgrade to 10.2.7, so I can't provide a functional test. 

Note that I've written some docs (that cover this question too) in the description. Point 6. 

Let me know if ok for you ?